compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1254,6 +1266,38 @@ public void visit(BLangBinaryExpr binaryExpr) {
         }
     }
 
+    /**
+     * This method checks whether given binary expression is related to byte type operation for the following
+     * conditions. If its true, then both lhs and rhs of the binary expression will be converted to 'int' type.
+     *
+     *      byte a = 12;
+     *      byte b = 34;
+     *      int i = 234;
+     *      int j = -456;
+     *
+     *      false: where binary expression's expected type is 'byte'
+     *      byte b1 = a | b;
+     *      byte b2 = a & b;
+     *      byte b3 = a ^ b;
+     *
+     *      true: where binary expression's expected type is 'int'
+     *      int i1 = a | i;
+     *      int i2 = a & i;
+     *      int i3 = a ^ i;
+     *      int i4 = a + b;
+     *      int i5 = a - b;
+     *      int i6 = a >> b;
+     *      int i7 = a << b;
+     *
+     */
+    private boolean isByteRelatedOperation(BLangBinaryExpr binaryExpr) {
```

Comment:
We don't need to handle this case especially, Typechecker should handle this case. check whether there is a bug.